python_modules/dagster/dagster_tests/storage_tests/utils/event_log_storage.py

```diff
@@ -452,6 +454,13 @@ def event_log_storage(self, request):
     def instance(self, request) -> Optional[DagsterInstance]:
         return None
 
+    @pytest.fixture
+    def end_instance(self, instance):
```

Comment:
this is confusing to me...  Why are they different?

Reply:
See the internal PR; but the essential thing is that in the graphql event log storage tests, the "instance" fixture is expected to be the host instance, rather than the user code instance. So we need a separate instance for the "user code instance" for the cases where they are different.

Reply:
I think the reason this is confusing is that we are presumably testing the event log storage implementation.

But we don't make use of the fixtured storage at all in the test.  We're effectively testing the implementation of `DagsterInstance` rather than any particular implementation of the `EventLogStorage`.

Instead, I would expect to see the following:

```
def test_get_asset_check_state(self, storage: EventLogStorage):
    ...
    events = _synthesize_events(...)
    for event in events:
        storage.store_event(event)
    latest_check_info = storage.get_latest_check_info(...)
    # assert things about the latest check info here
```

This way, we don't have to touch the instance at all, don't have to be aware that there is a storage class on the instance or anything.

Reply:
The nice thing about this is that we're really just testing the relationship between `store_event` and `get_asset_check_state` / `get_latest_check_info` / whatever we decide to call this.